### PR TITLE
Refactor repetitive calls

### PR DESCRIPTION
**Pull request by schochastics**
This PR fixes the repetitive calls in all API calls with a function `process_request()` 

It also fixes some path errors in 
`get_account_bookmarks()` `get_account_favourites()` `get_account_mutes()` `get_account_blocks()`

Additionally, it exposes ratelimit and paging parameters as attributes

@chainsawriot happy to not merge if you think this interferes with your test setup too much. I did not run into problems so far
Your call!

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -10,9 +10,56 @@ print.rtoot_bearer <- function(x,...){
...
+                FUN = identity
+){
+  output <- make_get_request(token = token,path = path,
+                             instance = instance, params = params,
+                             anonymous = anonymous)
+  if (isTRUE(parse)) {
+    header <- attr(output,"headers")
+    if(length(output)<1){
+      output <- tibble::tibble()
+    } else if(length(output[[1]])==1){ #TODO:fragile?
```
Comment (chainsawriot) [inline R/utils.R:57]: This is indeed a bit fragile.

Comment (schochastics) [reply R/utils.R:57]: Yeah trying to work out a better solution. Maybe check for depth of list?
```r
# https://stackoverflow.com/a/13433689
depth <- function(this,thisdepth=0){
  if(!is.list(this)){
    return(thisdepth)
  }else{
    return(max(unlist(lapply(this,depth,thisdepth=thisdepth+1))))    
  }
} 
```
if depth =1 then we have only one object and if depth>1 we have a list of object?

Comment (chainsawriot) [reply R/utils.R:57]: I think a better way to handle this is to use some crazy functional programming technique.

```r
## vectorizing
v <- function(FUN) {
    v_FUN <- function(x) {
        dplyr::bind_rows(lapply(x, FUN))
    }
    return(v_FUN)
}
```

For the functions that need vectoring, pass `v(parse_status)` as `FUN` to `process_request`

**Comment by chainsawriot** [comment]
Other minor points you can decide

1. Should we still return `headers` if someone uses `parse = FALSE`?
2. I think it is also time to move `make_get_request` to `utils.R`.

**Comment by schochastics** [comment]
1. for now I think we should. We can kick that out later
2. yes thought that too and will do that

File: R/utils.R (hunk 1 above)
Comment (schochastics) [reply R/utils.R:91]: Holy! I try it out

Comment (schochastics) [reply R/utils.R:91]: but does it capture cases where we would usually expect several objects returned, but we only get one?

Comment (chainsawriot) [reply R/utils.R:91]: I think it applies only to `get_context`. For that case, we just need to craft a specialize `FUN` for it.

So:

| Function | FUN |
|-------------|--------|
| `get_status` | parse_status |
| `get_public_timeline` | v(parse_status) |
| `get_context` | a specialize FUN |

Comment (schochastics) [reply R/utils.R:91]: I test it

Comment (chainsawriot) [reply R/utils.R:91]: https://github.com/schochastics/rtoot/blob/89bed02988df595e3c767b661805450c39220b49/R/timelines_statuses.R#L97-L100

Something like

```r
parse_context <- function(output) {
    temp_output <- list()
    temp_output$ancestors <- dplyr::bind_rows(lapply(output$ancestors, parse_status))
    temp_output$descendants <- dplyr::bind_rows(lapply(output$descendants, parse_status))
     temp_output
}
```

**Comment by chainsawriot** [comment]
Nice! I think it is fine now.

**Pull request merged**